pkg/tfgen/docs.go

```diff
@@ -1201,52 +1203,54 @@ func (p *tfMarkdownParser) parseImports(subsection []string) {
 		}
 	}
 
-	if i, ok := tryParseV2Imports(token, subsection); ok {
+	if i, ok := tryParseV2Imports(token, body); ok {
 		p.ret.Import = i
 		return
 	}
 
-	var importDocString string
-	for _, section := range subsection {
-		if strings.Contains(section, "**NOTE:") || strings.Contains(section, "**Please Note:") ||
-			strings.Contains(section, "**Note:**") {
+	importDocs := new(bytes.Buffer)
+	for _, line := range strings.Split(body, "\n") {
+		if strings.Contains(line, "**NOTE:") || strings.Contains(line, "**Please Note:") ||
+			strings.Contains(line, "**Note:**") {
 			// This is a Terraform import specific comment that we don't need to parse or include in our docs
 			continue
 		}
 
 		// Skip another redundant comment
-		if strings.Contains(section, "Import is supported using the following syntax") {
+		if strings.Contains(line, "Import is supported using the following syntax") {
 			continue
 		}
 
 		// Remove the shell comment characters to avoid writing this line as a Markdown H1:
-		section = strings.TrimPrefix(section, "# ")
+		if l1, ok := strings.CutPrefix(line, "#"); ok {
```

Comment:
Isn't this doing the same as the previous TrimPrefix?

Reply:
No. The new logic will remove `#` if it constitutes the entire line *or* if it is followed by a space.

The logic would have been better expressed as:

```go
		if line == "#" {
			line = ""
		} else {
			line = strings.TrimPrefix(line, "# ")
		}
```